packages/flutter_tools/lib/src/asset.dart

```diff
@@ -487,42 +494,44 @@ Map<_Asset, List<_Asset>> _parseAssets(
   return result;
 }
 
-_Asset _resolvePackageAsset(
-    String assetBase,
-    String packageName,
-    String asset,
-) {
-  return new _Asset(
-    base: assetBase,
-    assetEntry: 'packages/$packageName/$asset',
-    relativePath: asset,
-  );
-}
-
 _Asset _resolveAsset(
   PackageMap packageMap,
   String assetBase,
   String asset,
+  String packageName,
 ) {
   if (asset.startsWith('packages/') && !fs.isFileSync(fs.path.join(assetBase, asset))) {
-    // Convert packages/flutter_gallery_assets/clouds-0.png to clouds-0.png.
-    String packageKey = asset.substring(9);
-    String relativeAsset = asset;
-
-    final int index = packageKey.indexOf('/');
-    if (index != -1) {
-      relativeAsset = packageKey.substring(index + 1);
-      packageKey = packageKey.substring(0, index);
-    }
+    final _Asset packageAsset = _resolvePackageAsset(asset, packageMap);
+    if (packageAsset != null)
+      return packageAsset;
+  }
 
-    final Uri uri = packageMap.map[packageKey];
-    if (uri != null && uri.scheme == 'file') {
-      final File file = fs.file(uri);
-      return new _Asset(base: file.path, assetEntry: asset, relativePath: relativeAsset);
-    }
+  final String assetEntry = packageName != null
```

Comment:
Maybe add a small comment on the difference between these two: Why do we need to resolve the first on (above), but not this one?